GoMeet/src/main/java/com/google/sps/servlets/MeetingTimeServlet.java

```diff
@@ -0,0 +1,118 @@
+  /** Fetches a MeetingTime entity from Datastore according to the entity ID in the query string */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String keyStr = request.getParameter(MeetingTimeFields.MEETING_TIME_ID);
+    if (keyStr == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.BAD_REQUEST_ERROR);
+      return;
+    }
+
+    Key key;
+    try {
+      key = KeyFactory.stringToKey(keyStr);
+    } catch(IllegalArgumentException e) {
+      // the keyStr is an invalid key
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.INVALID_KEY_ERROR);
+      return;
+    }
+    
+    // filter by Key
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity result;
+
+    try {
+      result = datastore.get(key);
+    } catch (EntityNotFoundException e) {
+      // entity by the given key is not found
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_NOT_FOUND, ErrorMessages.ENTITY_NOT_FOUND_ERROR);
+      return;
+    }
+    
+    // Each of the fields will be set to null if not found in returned entity
+    String datetime = (String) result.getProperty(MeetingTimeFields.DATETIME);
+    Long voteCount = 
+        result.getProperty(MeetingTimeFields.VOTE_COUNT) == null ? 
+        null : (long) result.getProperty(MeetingTimeFields.VOTE_COUNT);
+    List voters = (ArrayList) result.getProperty(MeetingTimeFields.VOTERS);
+    // add all to hashmap,
+    HashMap<String, Object> meetingTime = new HashMap<String, Object>() {{
+      put(MeetingTimeFields.DATETIME, datetime);
+      put(MeetingTimeFields.VOTE_COUNT, voteCount);
+      put(MeetingTimeFields.VOTERS, voters);
+    }};
+
+    // return as JSON
+    String meetingTimeJson = ServletUtil.convertMapToJson(meetingTime);
+    response.getWriter().println(meetingTimeJson);
+  }
+  
+  /** Creates a new MeetingTime entity and stores it to Datastore */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // expect the query string to be of the following format: ?datetime=DATETIME
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String datetime = request.getParameter(MeetingTimeFields.DATETIME);
+
+    if (datetime == null) {
+      ServletUtil.sendErrorResponse(response, HttpServletResponse.SC_BAD_REQUEST, ErrorMessages.BAD_REQUEST_ERROR);
+      return;
+    }
+
+    Entity meetingTime = new Entity("MeetingTime");
+    meetingTime.setProperty(MeetingTimeFields.DATETIME, datetime);
+    meetingTime.setProperty(MeetingTimeFields.VOTE_COUNT, 0); // initially votes are 0
+    datastore.put(meetingTime);
+    String meetingTimeKey = KeyFactory.keyToString(meetingTime.getKey());
```

Comment:
Just a question - what do you do about meeting times that have been created but might be orphaned because the user session was interrupted between when the meeting time was created and when it was attached to the meeting?

Reply:
(Is that a valid question)

Reply:
Hmm this is a valid question - I've googled around and potentially, we could associate the meeting time as a child entity of the meeting entity, and then routinely (e.g. every time the app launches) loop through every MeetingTime, checking whether their parent meeting exists, and delete them if the parent doesn't exist. 
There doesn't seem to be a way to easily cascade delete orphaned entities, it would need to be done by looping through and checking them all.

Reply:
One possibility could be to adjust the state flow so that you would associate a meeting with every meeting time when the meeting time is created, but that might require some reworking to pass a meeting identifier around.